client/app/scripts/components/timeline-control.js

```diff
@@ -0,0 +1,251 @@
+      this.setState({ millisecondsInPast: rangeMilliseconds });
+      this.updateTimestamp(rangeMilliseconds);
+      this.props.startWebsocketTransition();
+    }
+  }
+
+  handleJumpToNowClick() {
+    this.setState({
+      showSliderPanel: false,
+      millisecondsInPast: 0,
+      rangeOptionSelected: sliderRanges.last1Hour,
+    });
+    this.updateTimestamp();
+    this.props.startWebsocketTransition();
+  }
+
+  handleTimestampClick() {
+    this.setState({ showSliderPanel: !this.state.showSliderPanel });
+  }
+
+  updateTimestamp(millisecondsInPast = 0) {
+    this.props.websocketQueryInPast(millisecondsInPast);
+    this.props.clickResumeUpdate();
+  }
+
+  getRangeMilliseconds(rangeOption = this.state.rangeOptionSelected) {
+    return moment().diff(rangeOption.getStart());
+  }
+
+  renderRangeOption(rangeOption) {
+    const handleClick = () => { this.handleRangeOptionClick(rangeOption); };
+    const selected = (this.state.rangeOptionSelected.label === rangeOption.label);
+    const className = classNames('option', { selected });
+
+    return (
+      <a key={rangeOption.label} className={className} onClick={handleClick}>
+        {rangeOption.label}
+      </a>
+    );
+  }
+
+  renderJumpToNowButton() {
+    return (
+      <a className="button jump-to-now" title="Jump to now" onClick={this.handleJumpToNowClick}>
+        <span className="fa fa-step-forward" />
+      </a>
+    );
+  }
+
+  renderTimelineSlider() {
+    const { millisecondsInPast } = this.state;
+    const rangeMilliseconds = this.getRangeMilliseconds();
+
+    return (
+      <Slider
+        onChange={this.handleSliderChange}
+        value={rangeMilliseconds - millisecondsInPast}
+        max={rangeMilliseconds}
+      />
+    );
+  }
+
+  render() {
+    const { websocketTransitioning, hasTimelineControl } = this.props;
+    const { showSliderPanel, millisecondsInPast } = this.state;
+    const isCurrent = (millisecondsInPast === 0);
+
+    // Don't render the timeline control if it's not explicitly enabled for this instance.
+    if (!hasTimelineControl) return null;
+
+    return (
+      <div className="timeline-control">
+        {showSliderPanel && <div className="slider-panel">
+          <span className="caption">Explore</span>
+          <div className="options">
+            <div className="column">
+              {this.renderRangeOption(sliderRanges.last15Minutes)}
+              {this.renderRangeOption(sliderRanges.last1Hour)}
+              {this.renderRangeOption(sliderRanges.last6Hours)}
+              {this.renderRangeOption(sliderRanges.last24Hours)}
+            </div>
+            <div className="column">
+              {this.renderRangeOption(sliderRanges.last7Days)}
+              {this.renderRangeOption(sliderRanges.last30Days)}
+              {this.renderRangeOption(sliderRanges.last90Days)}
+              {this.renderRangeOption(sliderRanges.last1Year)}
+            </div>
+            <div className="column">
```

Comment:
I think I'll keep it for now.. Chances are all these columns will go away anyway in the next push of the UI changes to the time travel component (which will happen soon hopefully).